### ISSUE TITLE
Strange behaviour with prometheus/alertmanager alert transitions

### ISSUE THREAD

**Issue by hasso** [guardian/alerta#189]
Here is important snippets from my configurations. Prometheus.rules:

```
ALERT alertName
    IF (metric > 1000) FOR 1m
    LABELS { severity="critical" }
    ANNOTATIONS {
    }

ALERT alertName
    IF (metric > 100) FOR 1m
    LABELS { severity="major" }
    ANNOTATIONS {
    }
```

Alertmanager.yml:

```
route:
  receiver:        'alerta'
  group_wait:      10s
  group_interval:  10s
  repeat_interval: 60s
  group_by:        ['alertname', 'host']

inhibit_rules:
- source_match:
    severity: 'critical'
  target_match:
    severity: 'major'
  equal: ['alertname', 'host']

receivers:
- name: 'alerta'
  webhook_configs:
  - url: 'http://localhost:8080/webhooks/prometheus?api-key=key'
    send_resolved: true
```

From alertmanager seems that all is working as expected, but there is a strange behaviour in Alerta.
 * If metric rises over 100, the major alarm is firing in alertmanager. Alerta receives the alert correctly.
 * If metric rises over 1000 , the critical alarm is firing in alertmanager and the major is inhibited. Alerta updates the alert correctly.
 * If metric falls under 1000 again (but is still over 100) alertmnager removes the critical one and the major one is firing again. But Alerta gets confused for some reason - number of duplicates is zeroed, it's sometimes unclear whether its major or critical now, and duplicate updates (different ID, but same datetime etc) will appear under alert data.

**Comment by satterly** [comment]
Thanks. I'll see if I can reproduce the issue. I suspect the problem is on the Prometheus side not on the Alerta side but we'll see

**Comment by hasso** [comment]
Two comments which might be helpful.

* If send_resolved is enabled and an alert is transitioning between two severities with configuration like this (inhibit_rule and send_resolved set), there is indeed inconsistent behaviour from alertmanager. In case of major -> critical transition it sends notification for critical state only. But in case of critical -> major transition it sends OK for critical at first and then fires major one again.
* Update timestamps for these transitions are set to wrong timestamp IMHO - database.py line 450. Why not now?

**Comment by satterly** [comment]
Thanks for looking into this further. It sounds like this is definitely an issue that should be raised with Prometheus. Do you want me to do it or will you?

On the second point, you a right that the update timestamps should not be set to `createTime`. They should be set to `lastReceiveTime` which is, as you suggest, `now`.

**Comment by satterly** [comment]
The update timestamp fix is available now as version 4.7.18.

**Comment by hasso** [comment]
I discussed the issue briefly with prometheus developers in irc and they agreed that there is a problem with inconsistency, but somehow concluded that it's a major -> critical transition that needs to be "fixed". Ie for every transition you'll get an OK for old state, and then a new notification for a new state. I'm not in position to argue with this, so it's probably better if you do it.

**Comment by satterly** [comment]
Hmm. An OK between every transition? I've never heard of that before. That doesn't sound right but I'd need to hear from them why they think it should work like that. I'll see what I can find out. Thanks.

**Comment by satterly** [comment]
This issue could have been resolved by the fix #234 . Any chance you could update to the latest alerta server version to test and let me know?
